lib/features/details/presentation/view/movie_details_page.dart

```diff
@@ -18,10 +20,48 @@ class MovieDetailsPage extends StatefulWidget {
 class _MovieDetailsPageState extends State<MovieDetailsPage> {
   DetailsViewModel detailsViewModel = DetailsViewModel();
 
+  ValueNotifier<bool> _movieInWatchList = ValueNotifier(false);
+  ValueNotifier<bool> get movieInWatchList => _movieInWatchList;
```

Comment:
You don't need this getter.
No other class apart from `_MovieDetailsPageState` needs the `_movieInWatchList` so you're good with just the private instance.

The reason you use getters in your view models is to keep the instance private so it can't be changed from outside. But at the same time, exposing it to your widget that might need it. In this case, other classes need the notifier so you expose them using getters in a way that allows them access the notifier but not modify it.